Add local event docs (#399)
holochain/docs-pages: src/pages/get-building/at-an-event.md

---

> title: Getting Set Up For a Live Event

Reviewer: ```suggestion
title: Getting Set Up For a Local Event
```

---

## Prepare your development environment

> If you have not already done so, please follow the [Quick Start](/quick-start/) guide. The rest of this guide assumes you have either done this or created an equivalent environment manually.

Reviewer: ideally they would run the `setup.sh` step. then come back here to setup the local caching, and then go there again to run the `nix run -j0 ...#hc-scaffold`, right?

Author: Preferably yes

Author: Updated

---
title: Getting Set Up For a Local Event
---

## Prepare your development environment

If you have not already done so, please follow the first step in [Quick Start](/quick-start/) guide which asks you to run the `setup.sh` script. Then you should come back to this page and configure the local cache before proceeding with verifying your Nix setup in the quick start guide.

The rest of this guide assumes you have either done this or created an equivalent environment manually.

## Configure Nix to use a local cache

> !!! note Pre-requisite
> Please check with the event hosts whether a local cache is being provided. If not, this section can be skipped.
> !!!

Reviewer 2: If there's no local cache then there's no point to follow this whole tutorial, is there?
```suggestion
```

Reviewer 2: ah okay. Yes, I prefer that. I was confused by what to skip without a local cache.

> The quick start `setup.sh` script configures Nix to use `https://holochain-ci.cachix.org` as a cache. This significantly speeds up launching the Holonix development environment. It works well in general, but at live events it requires a lot of bandwidth with multiple people downloading large files at the same time. To improve the experience for everyone, a cache can be provided locally. This section shows you how to connect to the local cache.

Reviewer 2: ```suggestion
The quick start `setup.sh` script configures Nix to use `https://holochain-ci.cachix.org` as a cache. This significantly speeds up launching the Holonix development environment. It works well for individuals, but at live events it requires a lot of bandwidth with multiple people downloading large amounts of files at the same time. To improve the experience for everyone, a cache can be provided locally. This section shows you how to connect to the local cache.
```

> The event hosts will provide you with a cache name. This will be unique per event! Replace `<event-cache-name>` with the provided name in the command below.

Reviewer 2: ```suggestion
The event hosts need to provide you with a cache name. This will be unique per event! Replace `<event-cache-name>` with the provided name in the command below.
```
